### PR TITLE
Update codebase to PHP 7.4

### PR DESCRIPTION
**Pull request by TavoNiievez**

### CODE REVIEW

File: AssertThrows.php [hunk 1]
```diff
@@ -32,12 +30,12 @@ private function checkException(?string $message, Throwable $exception)
...
      * Asserts that callback throws an exception
      *
-     * @param $throws
+     * @param string|Throwable $throws
      * @param callable $func
      * @param mixed ...$params
      * @throws Throwable
      */
-    public function assertThrows($throws, callable $func, ...$params)
+    public function assertThrows($throws, callable $func, array $params = [])
```
Comment (boboudreau) [inline AssertThrows.php:38]: Might this be backward incompatible to many use cases? If any number of arguments are passed into a variable length argument list, PHP treats it as an array by default. How does this fix PHP 7.4 compatibility?

Wondering what this broke and why my team might need to update all of our tests... 😄 

Would be happy to put in a PR to revert this, if that's acceptable.

Comment (TavoNiievez) [reply AssertThrows.php:38]: @boboudreau you can send a PR for this.

Comment (boboudreau) [reply AssertThrows.php:38]: I should have explicitly stated the issue; there are some tests in which we want to call a method with multiple arguments, but have passed them as the 4th, 5th, 6th, etc. argument to this method, and not as an array. 

I see that this was at least versioned as 1.2 vs. 1.1 but generally there shouldn't be code changes that cause Errors that are Throwable (in this case, a TypeError) in a minor version release (according to semver), correct?

Comment (TavoNiievez) [reply AssertThrows.php:38]: is correct, please include how you suggest your change be tagged in your PR description.

**Pull request merged**